pkg/kube/controllers/quarkssecret/quarkssecret_controller.go

```diff
@@ -97,6 +102,64 @@ func AddQuarksSecret(ctx context.Context, config *config.Config, mgr manager.Man
 		return errors.Wrapf(err, "Watching quarks secrets failed in quarksSecret controller.")
 	}
 
+	// Watch for changes to user created secrets
+	p = predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return false },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+		GenericFunc: func(e event.GenericEvent) bool { return false },
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			n := e.ObjectNew.(*corev1.Secret)
+			o := e.ObjectOld.(*corev1.Secret)
+
+			shouldProcessReconcile := isUserCreatedSecret(n)
+			if reflect.DeepEqual(n.Data, o.Data) && reflect.DeepEqual(n.Labels, o.Labels) &&
+				reflect.DeepEqual(n.Annotations, o.Annotations) {
+				return false
+			}
+
+			return shouldProcessReconcile
+		},
+	}
+	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
+			secret := a.Object.(*corev1.Secret)
+
+			if skip.Reconciles(ctx, mgr.GetClient(), secret) {
+				return []reconcile.Request{}
+			}
+
+			reconciles, err := reference.GetReconciles(ctx, mgr.GetClient(), reference.ReconcileForQuarksSecret, secret, false)
+			if err != nil {
+				ctxlog.Errorf(ctx, "Failed to calculate reconciles for secret '%s/%s': %v", secret.Namespace, secret.Name, err)
+			}
+
+			for _, reconciliation := range reconciles {
+				ctxlog.NewMappingEvent(a.Object).Debug(ctx, reconciliation, "QuarksSecret", a.Meta.GetName(), qsv1a1.KubeSecretReference)
+
+				qsec := qsv1a1.QuarksSecret{}
+				err := mgr.GetClient().Get(ctx, types.NamespacedName{Name: reconciliation.Name, Namespace: reconciliation.Namespace}, &qsec)
+				if err != nil {
+					ctxlog.Errorf(ctx, "could not get QuarksSecret '%s': %v", qsec.GetNamespacedName(), err)
+				}
+
+				if qsec.Status.Generated == nil {
+					qsec.Status = qsv1a1.QuarksSecretStatus{}
+				}
+				qsec.Status.Generated = pointers.Bool(false)
+				qsec.Status.LastReconcile = nil
+				err = mgr.GetClient().Status().Update(ctx, &qsec)
```

Comment:
This update triggers another reconcile on the qsec?

Reply:
Changed the logic